lib/Backend/GlobOpt.cpp

```diff
@@ -7184,13 +7198,36 @@ GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
     if (prevSymStore && prevSymStore->IsStackSym() &&
         prevSymStore->AsStackSym()->HasByteCodeRegSlot())
     {
-        Assert(this->blockData.changedSyms != nullptr);
-        this->blockData.changedSyms->Set(prevSymStore->m_id);
+        this->SetChangedSym(prevSymStore->m_id);
     }
     valueInfo->SetSymStore(sym);
 }
 
 void
+GlobOpt::SetChangedSym(SymID symId)
+{
+    GlobOptBlockData * globOptData = nullptr;
+    if (this->currentBlock->globOptData.changedSyms)
+    {
+        globOptData = &this->currentBlock->globOptData;
+    }
+    else
```

Comment:
Yes, this is only for MergeValues so far. MergeValues could create new value for merge, and merge changed syms for building bailout doesn't handle that. Do we need to be conservative to track these?

---
In reply to: [72355295](https://github.com/Microsoft/ChakraCore/pull/1337#discussion_r72355295) [](ancestors = 72355295)